src/peft/tuners/adalora/layer.py

```diff
@@ -73,32 +82,31 @@ def reset_lora_parameters(self, adapter_name):
             nn.init.normal_(self.lora_B[adapter_name], mean=0.0, std=0.02)
 
 
-class SVDLinear(nn.Linear, AdaLoraLayer):
+class SVDLinear(nn.Module, AdaLoraLayer):
     # SVD-based adaptation by a dense layer
     def __init__(
         self,
+        base_layer: nn.Module,
```

Comment:
Same comment above about BC, I would slowly deprecate them instead of changing the signature of these public classes